thavalon-server/src/database/accounts/mod.rs

```diff
@@ -2,22 +2,64 @@ pub mod account_errors;
 use super::get_database;
 use account_errors::AccountError;
 use mongodb::{
-    bson::{self, doc},
+    bson::{self, doc, oid::ObjectId, Document},
     options::{FindOneOptions, UpdateOptions},
 };
 use serde::{Deserialize, Serialize};
 const USER_COLLECTION: &str = "thavalon_users";
+
 /// Canonical representation of a database account.
 /// This should never leave the database, as it contains a password hash!
-#[derive(Serialize, Deserialize)]
 pub struct DatabaseAccount {
+    pub id: String,
     pub email: String,
     pub hash: String,
     pub display_name: String,
     pub profile_picture: Option<Vec<u8>>,
     pub email_verified: bool,
 }
 
+/// Internal representation of a user. Notably, this uses _id as an ObjectId
+/// instead of a hex string. This also supports serialization and deserialization.
+#[derive(Serialize, Deserialize)]
+struct InternalDBAccount {
+    _id: ObjectId,
+    email: String,
+    hash: String,
+    display_name: String,
+    profile_picture: Option<Vec<u8>>,
+    email_verified: bool,
+}
+
+impl From<DatabaseAccount> for InternalDBAccount {
```

Comment:
Maybe we should implement `TryFrom` instead? That way, if we ever end up with an invalid ID, like from user input, it won’t crash the server

Reply:
Fixed